packages/cli/src/bin.ts

```diff
@@ -164,63 +153,21 @@ async function main(): Promise<number> {
   }
   console.log(result)
 
-  try {
-    // SHA256 identifier for the project based on the Prisma schema path
-    const projectPathHash = await getProjectHash()
-    // SHA256 of the cli path
-    const cliPathHash = getCLIPathHash()
-
-    let schemaProvider: string | undefined
-    let schemaPreviewFeatures: string[] | undefined
-    let schemaGeneratorsProviders: string[] | undefined
-    try {
-      const schema = await getSchema(args['--schema'])
-      const config = await getConfig({
-        datamodel: schema,
-        ignoreEnvVarErrors: true,
-      })
-
-      if (config.datasources.length > 0) {
-        schemaProvider = config.datasources[0].provider
-      }
-
-      // restrict the search to previewFeatures of `provider = 'prisma-client-js'`
-      // (this was not scoped to `prisma-client-js` before Prisma 3.0)
-      const generator = config.generators.find(
-        (generator) => parseEnvValue(generator.provider) === 'prisma-client-js' && generator.previewFeatures.length > 0,
-      )
-      if (generator) {
-        schemaPreviewFeatures = generator.previewFeatures
-      }
-
-      // Example 'prisma-client-js'
-      schemaGeneratorsProviders = config.generators.map((generator) => parseEnvValue(generator.provider))
-    } catch (e) {
-      debug('Error from cli/src/bin.ts')
-      debug(e)
-    }
-
-    // check prisma for updates
-    const checkResult = await checkpoint.check({
-      product: 'prisma',
-      cli_path_hash: cliPathHash,
-      project_hash: projectPathHash,
-      version: packageJson.version,
-      schema_providers: schemaProvider ? [schemaProvider] : undefined,
-      schema_preview_features: schemaPreviewFeatures,
-      schema_generators_providers: schemaGeneratorsProviders,
-      cli_path: process.argv[1],
-      cli_install_type: isPrismaInstalledGlobally ? 'global' : 'local',
-      command: commandArray.join(' '),
-      information: args['--telemetry-information'] || process.env.PRISMA_TELEMETRY_INFORMATION,
-    })
-    // if the result is cached and we're outdated, show this prompt
-    const shouldHide = process.env.PRISMA_HIDE_UPDATE_MESSAGE
-    if (checkResult.status === 'ok' && checkResult.data.outdated && !shouldHide) {
-      printUpdateMessage(checkResult)
-    }
-  } catch (e) {
-    debug(e)
+  /**
+   * Prepare data and run the Checkpoint Client
+   * See function for more info
+   */
+  const checkResult = await runCheckpointClientCheck({
+    commandArray,
+    isPrismaInstalledGlobally,
+    schemaPath: args['--schema'],
+    telemetryInformation: args['--telemetry-information'],
+    version: packageJson.version,
```

Comment:
How did you decide what version is read inside `runCheckpointClientCheck` vs. here? 
Would `packageJson.version` not also make sense to read inside?

Reply:
Since we use the version for the error reporting there as well, it's easier to pass it around and also nicer to not read it from too many places.

Reply:
🤷